### PR TITLE
Data Connections: Create a new top-level page

### LINKED ISSUE #49904: Create a new page for "Connections"
### What is the goal?
We would like to create a new page, possibly called "Data Connections" that groups the possible ways of extending and connecting Grafana to external sources of data (e.g. data sources, app plugins, integrations).

In the beginning it is only going to be accessible by using a feature flag.

[Figma designs](https://www.figma.com/file/2L3Fqq7D2iUQWbS1B1uqTJ/Connections?node-id=4%3A40)
[Slack channel](https://raintank-corp.slack.com/archives/C034LMXLXQU)

### PR DESCRIPTION
**Pull request by leventebalogh**
**Related issue:** https://github.com/grafana/grafana/issues/49904 

### What changes?
It basically creates a new top-level page called "Data Connections" (can still change) **behind a feature flag** which tries to serve as a single source of truth for extending Grafana through data sources or plugins. More info in the issue.

**The goal** of this Pull Request is to create the page as a wrapper and to provide a way for the Cloud Squad to load their Cloud Onboarding app plugin under the "Cloud integrations" tab. 

https://user-images.githubusercontent.com/9974811/172832136-3f87e3ab-1b4c-43e1-9ec8-1db87bd6c4ed.mov

---

**Notes for the reviewer**
The new `<AppPluginLoader />` component is currently having functionality that is redundant to the `<AppRootPage />`.
On the other hand I find the new component more concise, and as far as I tested it also solves the issue of re-mounting the app component on nav-model changes, however in a bit simpler way (without using portals). This still needs more testing, just like covering some edge cases, but if everything works as expected I am planning to reuse the `<AppPluginLoader />` inside the `<AppRootPage />` in a backward compatible way.

Fixes #49904

### CODE REVIEW

**Review by mikkancso — APPROVED** [review]
Nice job! LGTM

**Review by marefr — APPROVED** [review]
LGTM nice and smooth great work :rocket: Mostly looked at the backend changes. This can be merged since behind a feature toggle.

File: public/app/features/data-connections/constants.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// The ID of the app plugin that we render under that "Cloud Integrations" tab
+export const CLOUD_ONBOARDING_APP_ID = 'grafana-easystart-app';
+
+// The ID of the main nav-tree item (the main item in the NavIndex)
+export const ROUTE_BASE_ID = 'data-connections';
+
+export enum ROUTES {
```
Comment (jackw) [inline public/app/features/data-connections/constants.ts:7]: Should this live in `types.ts`?

**Review by jackw — APPROVED** [review]
Awesome work as always @leventebalogh 👏

File: public/app/features/data-connections/constants.ts (hunk 1 above)
Comment (leventebalogh) [reply public/app/features/data-connections/constants.ts:7]: Thanks mate, good point, those enums should live in the same place. 👍 

I just realised thought that the `types.ts` file is actually not referenced by any module, so will get rid of it for now.

**Pull request merged**